### DISCUSSION TITLE
Hydrate and de-hydrate workflow state without a backing store

### DISCUSSION THREAD

**Discussion by jan-johansson-mr** [elsa-workflows/elsa-core#4551] [Q&A] [answered]
Hi there,

I get the impression that the engine is dependent on a store connection for storing and reading workflow state. Is it possible to encapsulate the hydrate and de-hydrate workflow state so that I can pass the state as a serialized context and thus not be dependent on an actual store powering the engine?

Kinds regards

**Comment by sfmskywalker** [comment, marked as answer]
Absolutely! Elsa 3 is architected in layers. At the lowest layer, you have the capability to handle the de-hydration and re-hydration of the workflow state. At this layer, persistence stores are not involved. Here's a brief example to demonstrate:

```csharp
// Execute the workflow.
var result = _workflowRunner.RunAsync(workflow);
var workflowState = result.WorkflowState;

// Serialize the workflow state (also known as workflow instance).
var json = await _workflowStateSerializer.SerializeAsync(workflowState);

// Deserialize the serialized workflow state.
var deserializedWorkflowState = await _workflowStateSerializer.DeserializeAsync(json);

// Resume the workflow using the deserialized state.
var resumeResult = _workflowRunner.RunAsync(workflow, deserializedWorkflowState);
```

I hope this helps! If you have further questions, please let us know.
